tests/fake-block-tracker.ts

```diff
@@ -9,9 +9,10 @@ import { JsonRpcEngine } from '@metamask/json-rpc-engine';
 export class FakeBlockTracker extends PollingBlockTracker {
   #latestBlockNumber = '0x0';
 
-  constructor() {
+  constructor({provider}: {provider: SafeEventEmitterProvider}) {
     super({
-      provider: new SafeEventEmitterProvider({ engine: new JsonRpcEngine() }),
```

Comment:
Here the `SafeEventEmitterProvider` gets replaced and tests making use of the `FakeBlockTracker` now have to supply a `provider` explicitly.